### ISSUE TITLE
Concealing at EOL appends previous line

### ISSUE THREAD

**Issue by zsugabubus** [vim/vim#6302]
**Describe the bug**

Concealing on an empty line prints previous line once again after conceal char.

**To Reproduce**

1. Run `vim --clean +'set cole=2 cocu=ni' +'normal OHello' +"call matchaddpos('Conceal', [[2, 1, 1]], 99, -1, {'conceal': 'x'})"`.
2. Should see:
```
Hello
xHello
```

**Expected behavior**

Something like:
```
Hello
x
```

**Environment (please complete the following information):**
 - Vim version: 8.2 Included patches: 1-1017 (compiled from master)
 - OS: Arch Linux

**Comment by lacygoill** [comment]
Regression introduced by https://github.com/vim/vim/commit/4d585022023b96f6507e8cae5ed8fc8d926f5140.

**Comment by brammool** [comment]
@ichizok

**Comment by zsugabubus** [comment]
Hmmm. Sorry, it seems yesterday I was a bit inaccurate. It happens on every line if I put a concealed match at EOL. (Do not know why I thought it can happen only with empty lines…)

**Comment by zsugabubus** [comment]
Just a quick question: Is a conceal allowed at EOL? How that supposed to work with `'listchars'`? For example, a concealed "\n" with `'nolist'` should make `'listchars'`'s “eol” hidden?

https://github.com/vim/vim/blob/df069eec3b90401e880e9b0e258146d8f36c474d/src/drawline.c#L1475-L1480

If I understand this comment correctly, a conceal at "\n" never supposed to be displayed, right? Or how should "past the end of the line" be interpreted?

---

I thought I messed up something with my patch but not. I found another related issue where Vim get stuck in an infinite loop… I thought I finally finished :| Plus here is an older issue: #4854.
